hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/ITestS3ACopyFromLocalFile.java

```diff
@@ -120,15 +120,48 @@ public void testCopyMissingFile() throws Throwable {
         () -> upload(file, true));
   }
 
+  /*
+   * The following path is being created on disk and copied over
+   * /parent/ (trailing slash to make it clear it's  a directory
+   * /parent/test1.txt
+   * /parent/child/test.txt
+   */
   @Test
-  @Ignore("HADOOP-15932")
-  public void testCopyDirectoryFile() throws Throwable {
-    file = File.createTempFile("test", ".txt");
-    // first upload to create
-    intercept(FileNotFoundException.class, "Not a file",
-        () -> upload(file.getParentFile(), true));
+  public void testCopyTreeDirectoryWithoutDelete() throws Throwable {
+    java.nio.file.Path srcDir = Files.createTempDirectory("parent");
```

Comment:
There's nothing here which verifies that the dest tree matches the source, including contents.

It needs something like `testCopyFile()` where the byte value of the two .txt files are compared with the values at the far end.